windows/shared/logging/logging-plog-windows/include/logging/windows/PlogDebugWrapper.hxx

```diff
@@ -0,0 +1,41 @@
+
+#ifndef PLOG_DEBUG_WRAPPER_HXX
+#define PLOG_DEBUG_WRAPPER_HXX
+
+#include <windows.h>
+
+#include <plog/Appenders/DebugOutputAppender.h>
+#include <plog/Appenders/DynamicAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+namespace logging
+{
+namespace plog
+{
+
+template <typename Formatter>
+class DebugWrapperAppender : public ::plog::DynamicAppender
+{
+public:
+    /**
+     * @brief Construct a new DebugWrapperAppender object
+     * This class adds a DebugOutputApender iff the DEBUG macro is set and IsDebuggerPresent() is true
+     *  otherwise this class just does the same as the input Appender
+     * @param providerHandle
```

Comment:
Looks like the c'tor comment needs updating. Also, we should probably move the comment about the general behavior of this class to the class declaration itself:

```suggestion
/**
 * Common class which routes log messages to an attached debugger for any operating system. 
 * Currently, only Windows is supported.
 */
template <typename Formatter>
class DebugWrapperAppender : public ::plog::DynamicAppender
{
public:
    /**
     * @brief Construct a new DebugWrapperAppender object
     * @param providerHandle
```